src/sync/discovery.ts

```diff
@@ -1,14 +1,93 @@
-import { Subscription } from "../models";
 import { getRepositorySummary } from "./jobs";
 import enhanceOctokit from "../config/enhance-octokit";
 import { Application } from "probot";
-import { Repositories, SyncStatus } from "../models/subscription";
-import {LoggerWithTarget} from "probot/lib/wrap-logger";
+import Subscription, { Repositories, Repository, SyncStatus } from "../models/subscription";
+import { LoggerWithTarget } from "probot/lib/wrap-logger";
 import { sqsQueues } from "../sqs/queues";
+import GitHubClient from "../github/client/github-client";
+import { getCloudInstallationId } from "../github/client/installation-id";
+import { DiscoveryMessagePayload } from "../sqs/discovery";
 
-export const DISCOVERY_LOGGER_NAME = "sync.discovery";
+/*
+* Mapping the response data into a map by repo.id as required by subscription.updateSyncState.
+*/
+const mapRepositories = (repositories: Repository[]): Repositories => {
+	return repositories.reduce((obj, repo) => {
+		obj[repo.id] = { repository: getRepositorySummary(repo) };
+		return obj;
+	}, {});
+};
+
+/*
+* Reset the sync count to zero.
+*/
+const resetSyncedReposCount = async (subscription: Subscription): Promise<void> => {
+	await subscription.updateSyncState({ numberOfSyncedRepos: 0 });
+};
+
+/*
+* Update the sync status of a batch of repos.
+*/
+const updateSyncState = async (subscription: Subscription, repositories: Repository[]): Promise<void> => {
+	const repos = mapRepositories(repositories);
+	await subscription.updateSyncState({
+		repos
+	});
+};
+
+/*
+* Checking the header atttribute link(type string[]) for a entry that contains "rel=next", this indicates more pages available.
+*/
+const checkHeaderForNextPage = (headers): boolean => {
+	const links = headers?.link?.split(",");
+	const regex = /(rel="next")/g;
+	const hasNextPage = links?.find(elem => elem.match(regex));
+	return hasNextPage;
```

Comment:
if links is `undefined`, the function just returns `undefined` or if not, it either finds the element and returns that, or doens't find it which is again, `undefined`.  I'm also pretty sure based on the documentation that you can simplify this function by just doing ```
checkHeaderForNextPage = (headers): boolean => !!headers.link?.includes(`rel="next"`);
```